OpenRA.Game/Server/Server.cs

```diff
@@ -20,106 +20,47 @@
 using OpenRA.Primitives;
 using OpenRA.Support;
 
-using XTimer = System.Timers.Timer;
-
 namespace OpenRA.Server
 {
-	public enum ServerState : int
+	public enum ServerState
 	{
-		WaitingPlayers = 1,
-		GameStarted = 2,
-		ShuttingDown = 3
+		WaitingPlayers, GameStarted, ShuttingDown
 	}
 
 	public class Server
 	{
+		#region Members
+		
+		public readonly ModData ModData;
+		public readonly ServerSettings Settings;
+		public readonly Session LobbyInfo;
 		public readonly IPAddress Ip;
 		public readonly int Port;
-
-		int randomSeed;
 		public readonly MersenneTwister Random = new MersenneTwister();
+		public readonly List<string> TempBans = new List<string>();
 
-		// Valid player connections
-		public List<Connection> Conns = new List<Connection>();
+		// Valid and pre-verified player connections
+		readonly List<Connection> conns = new List<Connection>();
+		readonly List<Connection> preConns = new List<Connection>();
+		readonly TypeDictionary serverTraits = new TypeDictionary();
+		readonly TcpListener listener;
 
-		// Pre-verified player connections
-		public List<Connection> PreConns = new List<Connection>();
+		public volatile ServerState State = new ServerState();
 
-		TcpListener listener = null;
+		public Map Map { get; set; }
 
-		TypeDictionary serverTraits = new TypeDictionary();
-		public Session LobbyInfo;
+		public IEnumerable<Connection> Connections { get { return conns.ToList(); } }
```

Comment:
Please check. If none of the usages need the collection in a specific order, remove the `.ToList()` from here.
If they do, maybe the method should return `List` instead of `IEnumerable` and the `.ToList()` is again redundant.